### PR TITLE
[metricbeat] fix ARN parsing in cloudwatch collector for API Gateway metrics

### PR DESCRIPTION
**Pull request by tommyers-elastic**
## What does this PR do?

Whilst investigating an issue around tag filtering not working for API Gateway metrics, it was discovered that the ARN parsing logic in the AWS cloudwatch collector did not correctly parse ARNs from cloudwatch which start with a leading `/`; this caused all API gateway metrics to get filtered out. The crux of this PR is pre-strip leading slashes (or colons, just in case) from the ARN resource string before splitting. The result is that ARN resource strings like `/apis/123`, `apis/123` both result in a 'short' identifier of `123`. 

This PR also removes an used field from the internal data structure `metricsWithStatistics`, which was causing some additional overhead when trying to grep the code. When updating the test case I added the 'happy path' tagging case to validate the difference between matching tags and not matching tags.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [ ]

## Related issues

Combined with https://github.com/elastic/beats/pull/32408 and https://github.com/elastic/beats/pull/32435, fixes https://github.com/elastic/beats/pull/32408.

### CODE REVIEW

File: x-pack/metricbeat/module/aws/utils_test.go [hunk 1]
```diff
@@ -238,7 +238,7 @@ func TestGetMetricDataPerRegion(t *testing.T) {
 	reqGetMetricData := mockSvc.GetMetricDataRequest(getMetricDataInput)
 	getMetricDataOutput, err := reqGetMetricData.Send(context.TODO())
 	if err != nil {
-		fmt.Println("failed getMetricDataPerRegion: ", err)
+		_ = fmt.Errorf("failed getMetricDataPerRegion: %w", err)
```
Comment (kaiyan-sheng) [inline x-pack/metricbeat/module/aws/utils_test.go:241]: nit: maybe we can also use `assert.NoError(t, err)` here instead.

File: x-pack/metricbeat/module/aws/utils_test.go [hunk 2]
```diff
@@ -405,6 +405,7 @@ func TestFindTimestamp(t *testing.T) {
 	}
 }
 
+//fixme: add test case.....
```
Comment (kaiyan-sheng) [inline x-pack/metricbeat/module/aws/utils_test.go:408]: can be removed now? 😄

File: x-pack/metricbeat/module/aws/utils_test.go (hunk 1 above)
Comment (tommyers-elastic) [reply x-pack/metricbeat/module/aws/utils_test.go:241]: ah thanks for this - i was blindly fixing linter errors and didn't stop to actually think about it.

**Review by kaiyan-sheng — APPROVED** [review]
Looks good to me!

**Pull request merged**